zeapp/server/build.gradle.kts

```diff
@@ -0,0 +1,48 @@
+plugins {
+    application
+    kotlin("jvm")
+    kotlin("plugin.serialization") version "1.9.0"
+}
+
+group = "de.berlindroid"
+version = "1.0-SNAPSHOT"
+
+repositories {
+    mavenCentral()
+    google()
+}
+
+dependencies {
+    implementation(project(":badge"))
+
+    implementation(libs.kotlinx.coroutines.core)
+    implementation(libs.ktor.core)
+    implementation(libs.ktor.tomcat)
+    implementation(libs.ktor.network.tls)
+    implementation(libs.ktor.content.negotiation)
+    implementation(libs.ktor.content.serialization)
+    implementation(libs.kotlinx.serialization.json)
+    implementation(libs.sl4j.simple)
+
+}
+
+application {
+    mainClass.set("de.berlindroid.zekompanion.server.Main")
+}
+
+tasks {
+    val fatJar = register<Jar>("fatJar") {
+        dependsOn.addAll(listOf("compileJava", "compileKotlin", "processResources")) // We need this for Gradle optimization to work
+        archiveClassifier.set("standalone") // Naming the jar
+        duplicatesStrategy = DuplicatesStrategy.EXCLUDE
+        manifest { attributes(mapOf("Main-Class" to application.mainClass)) } // Provided we set it up in the application plugin configuration
+        val sourcesMain = sourceSets.main.get()
+        val contents = configurations.runtimeClasspath.get()
+            .map { if (it.isDirectory) it else zipTree(it) } +
```

Comment:
A bit confusing, maybe `.takeIf` opportunity

Reply:
I got to admit, I'm not seeing how to make that expression nicer: If the dependency is a zip, it should unzip it, otherwise use it as is. I'm not seeing how the takeIf would help, because it either needs to be taken or converted, but not nulled.

Or, can you elaborate a bit more on it?